test/fw/ptl/utils/plugins/ptl_test_runner.py

```diff
@@ -173,6 +170,7 @@ def startTest(self, test):
         if self.handler not in ptl_logger.handlers:
             ptl_logger.addHandler(self.handler)
         self.handler.buffer.truncate(0)
+        self.handler.buffer.seek(0)
```

Comment:
will truncate(0) not do that?

Reply:
No. There is a change in behavior of StringIO.truncate() in Python 3 vs Python 2. 
In Python 2, when we do StringIO.truncate(0) it also does StringIO.seek(0). So it resets string buffer to size 0 and releases any memory previously allocated for the buffer. In Python 3, StringIO.truncate(0) doesn't release the memory. It just fills string buffer with '\0' and to release memory we have to explicitly call StringIO.seek(0) after StringIO.truncate(0).
There is a bug in Python3 bug tracker but they say it’s the expected behavior now. 
https://bugs.python.org/issue30250

Reply:
okay. Thanks.